### PR TITLE
Migrate current account getter from AccountUtils to UserAccountManager

### PR DESCRIPTION
**Pull request by ezaquarii**
The implementation delegates calls to old code, so there should
be no regressions in functionality.

This change is huge, because `AccountUtils.getCurrentOwnCloudAccount()`
is a magic call used almost everywhere.

Not everything is migrated as there are some deep calls of static helpers
that need conversion first: connectivity, themes and whats new.

Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

### CODE REVIEW

File: src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java [hunk 1]
```diff
@@ -690,7 +703,7 @@ private void parseShares(List<Object> objects) {
                 shares.add(ocShare);
 
                 // get ocFile from Server to have an up-to-date copy
-                RemoteOperationResult result = new ReadFileRemoteOperation(ocShare.getPath()).execute(mAccount,
+                RemoteOperationResult result = new ReadFileRemoteOperation(ocShare.getPath()).execute(account,
```
[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java]: False positive.

[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java]: False positive.

**Comment by AndyScherzinger** [comment]
Will need to leave this one to @tobiasKaminsky since I am currently without my IDE / laptop.

**Comment by ezaquarii** [comment]
@AndyScherzinger do you know if I can restart CI build? Emulator failed and I don't want to push a new commit every time this happens.

File: src/main/java/com/owncloud/android/providers/UsersAndGroupsSearchProvider.java [hunk 2]
```diff
@@ -105,12 +116,14 @@ public String getType(@NonNull Uri uri) {
...
     public boolean onCreate() {
+        AndroidInjection.inject(this);
+
         if (getContext() == null) {
             return false;
         }
 
         String AUTHORITY = getContext().getResources().getString(R.string.users_and_groups_search_authority);
-        ACTION_SHARE_WITH = getContext().getResources().getString(R.string.users_and_groups_share_with);
+        setActionShareWith(getContext());
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/providers/UsersAndGroupsSearchProvider.java]: Why did you changed this, but not AUTHORITY in line 125?

**Comment by tobiasKaminsky** [comment]
:+1: 
This is indeed a big change in terms of code, but as you pointed out no functional change. 
I really love it :heart:!

Thank you for this great effort!

I did a quick search to see where getCurrentOwncloudAccount(…) is still used:
image

In RichDocumentsWebView it should be possible to use the new AccountManager, as you already did in this class.

Same for app sort order in AppPreferencesImpl? As we use DI to instantiate it, we can also create AccountManager meanwhile, or?

**Comment by tobiasKaminsky** [comment]
If you base this PR onto https://github.com/nextcloud/android/pull/3895 then the diff to review would have been a lot less, or am I wrong?

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky might be able to help you with restarting builds. Afaik you should be able to just log into the ci server with your Github account and be able to restart builds.

**Comment by ezaquarii** [comment]
I thought it would be easy, but it creates a circular dependency (account manager requires settings, settings require account manager).

Breaking it would probably force us to change how we store account specific settings and I don't think we are ready for this yes.

PS Dagger will not allow you to have circular dependencies and will refuse to compile/generate code.

Cheers,
Chris

On April 15, 2019 6:57:24 AM UTC, Tobias 
>Same for app sort order in AppPreferencesImpl? As we use DI to
>instantiate it, we can also create AccountManager meanwhile, or? >

**Comment by tobiasKaminsky** [comment]
As discussed remaining migration will follow in upcoming PR.

**Comment by tobiasKaminsky** [comment]
Thanks again for this awesome contribution!

**Pull request merged**